legion/engine/physics/physx/systems/physx_physics_system.cpp

```diff
@@ -0,0 +1,60 @@
+#include "physx_physics_system.hpp"
+#include <physx/PxPhysicsAPI.h>
+
+namespace legion::physics
+{
+    using namespace physx;
+
+    PxDefaultAllocator		gAllocator;
+    PxDefaultErrorCallback	gErrorCallback;
+
+    PxFoundation* gFoundation = nullptr;
+    PxPvd* gPvd = NULL;
+
+    PxPhysics* gPhysics = NULL;
+
+    PxDefaultCpuDispatcher* gDispatcher = NULL;
+
+    //PhysX PVD Debugger Related
+    constexpr char* pvdHost = "127.0.0.1";
+    constexpr size_type defaultPVDListeningPort = 5425;
+    constexpr size_type defaultPVDHostTimeout = 10;
```

Comment:
I'd personally rather not have these as floating globals, I'd be fine having them as part of some static struct, but rather not as floating globals. Also for their initialization, it might be possible for multiple systems or engine instances to exist, and these probably should be shut down gracefully with the system when the system is shut down or restarted

Reply:
yeah most of these don't need to be globals. With the exception of PxPhysics as stated in the docs here.
![image](https://user-images.githubusercontent.com/37859544/155262130-0689ef2d-a405-45f7-ab68-177f5d95f91c.png)
Will change all the other globals into statics tho.

Reply:
Are you sure you're reading the documentation right here? Name mangling is only relevant for functions so to me this seems like they're referring to the function PxCreatePhysics() and not the resulting value

Reply:
Just tested it, no issues with making it static